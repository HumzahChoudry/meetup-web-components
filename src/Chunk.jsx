import React from 'react';
import cx from 'classnames';

export const CHUNK_CLASS = 'chunk';
/**
 * Design System Component: Provides `stripe` styled container for components
 * @module Chunk
 */
class Chunk extends React.Component {
	render() {
		const {
			children,
			className,
			...other
		} = this.props;

		const classNames = cx(
			CHUNK_CLASS,
			className
		);

		return (
			<div
				className={classNames}
				{...other}
			>
				{children}
			</div>
		);
	}
}

export default Chunk;
