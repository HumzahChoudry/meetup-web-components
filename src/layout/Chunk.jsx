// @flow

import React from 'react';
import cx from 'classnames';

import withLoading from '../utils/components/withLoading';

type Props = {|
	/** The child elements of the component */
	children: React$Node,
	className?: string,
	/** Whether the component is in a loading state */
	isLoading?: boolean,
	/** Props to pass to the `<Loading />` component */
	loadingProps?: {
		color?: string,
		scrimColor?: string,
		size?: MediaSizes,
	},
|};
/**
 * Design System Component: Provides `stripe` styled container for components
 * @module ChunkComponent
 */
export class ChunkComponent extends React.Component<Props> {
	render() {
		const {
			children,
			className,
			loadingProps = {}, // eslint-disable-line no-unused-vars
			isLoading,
			...other
		} = this.props;

		const classNames = cx('chunk', { 'component--isLoading': isLoading }, className);

		return (
			<div className={classNames} {...other}>
				{children}
			</div>
		);
	}
}

const Chunk = withLoading(ChunkComponent);
Chunk.displayName = 'Chunk';
export default Chunk;
