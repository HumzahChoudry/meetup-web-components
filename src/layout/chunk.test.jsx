import React from 'react';
import { shallow } from 'enzyme';

import { Chunk } from './Chunk';

const dangerousHTML = '<div>Dangerously setting inner HTML</div>';
const makeDanger = () => ({__html: dangerousHTML});

describe('Chunk', function() {
	const chunk = shallow(<Chunk />);

	it('exists', function() {
		expect(chunk).toMatchSnapshot();
	});

	it('can handle dangerouslySetInnerHTML', function() {
		const dangerousHTMLComponent = shallow(
			<Chunk
				dangerouslySetInnerHTML={makeDanger()}
			/>
		);
		expect(dangerousHTMLComponent.html()).toContain(dangerousHTML);
	});
});
