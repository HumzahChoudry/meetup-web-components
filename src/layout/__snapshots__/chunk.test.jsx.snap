// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chunk exists 1`] = `<Chunk />`;
